integration_tests/src/main/python/conditionals_test.py

```diff
@@ -470,3 +470,11 @@ def test_between_sql():
         lambda spark : gen_df(spark, _data_gen_for_between),
         "tab",
         sql)
+
+@pytest.mark.skipif(is_before_spark_400(), reason="Spark versions prior to Spark 400 do not support collate")
+@allow_non_gpu("ProjectExec", "Collate")
+def test_collate_contains_fallback():
+    data_gen = [("c1", string_gen), ("c2", string_gen)]
+    assert_gpu_fallback_collect(
+        lambda spark : gen_df(spark, data_gen).selectExpr("contains(collate(c1, 'UTF8_BINARY'), c2)"),
```

Comment:
I think we might be missing some tests here. We see that the collate expression falls back to the CPU, but I want to verify that other String Types also fall back to the CPU.

We want to verify that we only support StringType with NoConstraint and the Default collationId "UTF8_BINARY"

I think we can do this by creating a new table on the CPU where a string column is not supported and then verify that we fall back to the CPU if the GPU tries to read it.

Reply:
Added test cases about CharType and VarcharType, please refer to the PR description and code.
`TypeSig.isSupported` returns false when meets `CharType`.

Reply:
> We want to verify that we only support StringType with NoConstraint and the Default collationId "UTF8_BINARY"

This is true for our current Spark-Rapids code. TypeSig.isSupported returns false when meets CharType.

Reply:
Added more case, refer to the latest code. Now supports directly generating a StringGen with collation.